info.hiergiltdiestfu.aws.neptune.graphml/src/main/java/info/hiergiltdiestfu/aws/neptune/graphml/REST/NeptuneAdapter.java

```diff
@@ -1,223 +1,236 @@
-		final var ekey = entry.getKey();
-		final var eval = entry.getValue();
-
-		final KeyType k = o.createKeyType();
-		k.setId(HtmlUtils.htmlEscape(ekey));
-		k.setFor(forT);
-		// attr.name?
-		// attr.type
-		return k;
-	}
-
-	public ExtractedGraph extract() {
-		GraphTraversalSource g = setup();
-
-		var eg = extractGraph(g);
-		try {
-			g.close();
-		} catch (Exception t) {
-			System.err.println(t);
-		}
-
-		eg = analyzeAttributes(eg);
-
-		return eg;
-
-	}
-
-	private static ExtractedGraph extractGraph(GraphTraversalSource g) {
-
-		final var vortices = g.V().valueMap().with(WithOptions.tokens);
-		final var edges = g.E();// .valueMap().with(WithOptions.tokens);
-
-		List<Map<Object, Object>> vertList = new LinkedList<>();
-		while (vortices.hasNext()) {
-			vertList.add(vortices.next());
-		}
-		List<Map<Object, Object>> edgeList = new LinkedList<>();
-		while (edges.hasNext()) {
-			Map<Object, Object> result = new HashMap<>();
-			final var next = edges.next();
-			final var properties = next.properties();
-			result.put(SOURCE, next.outVertex().id());
-			result.put(TARGET, next.inVertex().id());
-			result.put(T.id, next.id());
-			result.put(T.label, next.label());
-			while (properties.hasNext()) {
-				final var prop = properties.next();
-				result.put(prop.key(), prop.value());
-			}
-			edgeList.add(result);
-		}
-
-		return new ExtractedGraph(vertList, edgeList);
-
-	}
-
-	private static ExtractedGraph analyzeAttributes(ExtractedGraph eg) {
-		final Map<String, Object> nodeProps = initProps(eg.getNodes()), edgeProps = initProps(eg.getEdges());
-
-		fillProps(nodeProps, eg.getNodes());
-		fillProps(edgeProps, eg.getEdges());
-
-		eg.setVertexProperties(nodeProps);
-		eg.setEdgeProperties(edgeProps);
-
-		return eg;
-	}
-
-	private static void fillProps(Map<String, Object> output, List<Map<Object, Object>> input) {
-		input.parallelStream().flatMap(m -> m.entrySet().stream()).filter(e -> !output.containsKey(e.getKey()))
-				.forEach(f -> output.put(f.getKey().toString(), f.getValue().getClass()));
-	}
-
-	private static Map<String, Object> initProps(List<Map<Object, Object>> list) {
-		if (list.isEmpty()) {
-			return Collections.emptyMap();
-		} else {
-			return new ConcurrentHashMap<>(list.get(0).size() * 3);
-		}
-	}
-
-	private static GraphTraversalSource setup() {
-		/*
-persistence.neptune.contactpoint=localhost
-persistence.neptune.contactport=8182
-persistence.neptune.certchainfile=resources/aws/SFSRootCAG2.pem
-		 */
-		
-		Cluster.Builder builder = Cluster.build();
-		builder.addContactPoint("localhost");
-		builder.port(8182);
-		builder.enableSsl(true);
-		builder.keyCertChainFile("resources/aws/SFSRootCAG2.pem");
-
-		Cluster cluster = builder.create();
-
-		GraphTraversalSource graph = traversal().withRemote(DriverRemoteConnection.using(cluster));
-//		final TinkerGraph graph = TinkerFactory.createGratefulDead();
-
-		return graph;
-	}
-}
+package info.hiergiltdiestfu.aws.neptune.graphml.REST;
```

Comment:
Erstmal: Warum ist der Package-Name nicht kleingeschrieben?
Dann: Der Neptun-Adapter gehört sicherlich nicht in das REST-Package? Aber weiter unten gibt's nochmal einen Kommentar zur Package-Struktur allgemein. Da müssen wir nochmal allgemein restrukturieren.